### DISCUSSION TITLE
Leetcode Problem Solvers

### DISCUSSION THREAD

**Discussion by agentm** [agentm/project-m36#283] [Show and tell]
Project:M36 user and contributor @YuMingLiao has written a series of blog posts on how he used leetcode problems to learn more about the relational algebra and TutorialD. Along the way, he also found some missing features in Project:M36 which we hope to address soon. It's worth a read!

[https://project-m36-fun.blogspot.com/](https://project-m36-fun.blogspot.com/)

**Comment by agentm** [comment]
Hi @YuMingLiao, in [your recent post](https://project-m36-fun.blogspot.com/2021/03/leetcode-184-department-highest-salary.html), you write:

```
:showexpr (enum :: A)
```

to show all the constructors of data type `A`. But what kind of enumeration would you expect to see for 

```
data A = A Text | B
```
? Or do you expect to enumerate the data constructors only and not have it operate recursively?

**Reply by YuMingLiao** [reply]
I don't think user will benefit from enum on large types. It is useful when finding a universe of a small type, say, constructors < 100.
I think enumerating data constructors only is useful enough. You can always cross product them later to find combinations.

**Reply by YuMingLiao** [reply]
Recursively enums....Hmm...If there is an infinite loop, then it is not meaningful to enum, I guess.
I don't understand the part of lazy relation.
